packages/components/src/internal/components/ontology/OntologyTreePanel.tsx

```diff
@@ -18,11 +19,30 @@ type PathNode = OntologyPath & TreeNode;
 
 interface OntologyTreeProps {
     root: PathModel;
-    onNodeSelection: (conceptCode: PathModel) => void;
+    onNodeSelection: (path: PathModel) => void;
+    alternatePath?: PathModel;
 }
 
 export const OntologyTreePanel: FC<OntologyTreeProps> = props => {
-    const { root, onNodeSelection } = props;
+    const { root, onNodeSelection, alternatePath } = props;
+    const [showLoading, setShowLoading] = useState<boolean>(false);
+    const fileTreeRef: RefObject<FileTree> = useRef();
+
+    // watch for changes to alternatePath so that we can make sure the tree data down to that node is loaded
+    useEffect(() => {
+        if (alternatePath?.path) {
+            fetchParentPaths(alternatePath.path).then(parentPaths => {
+                // if this is a deeply nested path, let's mask the tree with a loading message while we toggle parents
+                if (parentPaths.length > 5) setShowLoading(true);
+
+                toggleParentPaths(fileTreeRef.current, alternatePath, parentPaths, true, () => {
+                    setShowLoading(false);
+                    scrollToActive();
```

Comment:
The DOM element to scroll to will have to be rendered before we can find it an call `scrollIntoView` on it. Note that the `scrollToActive` function has an if statement to check for this case so that if the element isn't found, it waits a half second and then tries again.